File: onyx-database/src/com/onyx/exception/EntityTypeMatchException.java
package com.onyx.exception;

/**
 * Created by timothy.osborn on 12/6/14.
 *
 * Entity type does not fit expected type
 */
public class EntityTypeMatchException extends EntityException {

    public static final String ATTRIBUTE_TYPE_IS_NOT_SUPPORTED = "Attribute type is not supported";

    /**
     * Constructor
     *
     * @param message Error message
     */
    public EntityTypeMatchException(String message)
    {
        super(message);
    }

    @SuppressWarnings("unused")
    public EntityTypeMatchException()
    {
        super();
    }
}

File: onyx-database/src/main/kotlin/com/onyx/exception/EntityTypeMatchException.kt
package com.onyx.exception

/**
 * Created by timothy.osborn on 12/6/14.
 *
 * Entity type does not fit expected type
 */
class EntityTypeMatchException @JvmOverloads constructor(message: String? = "") : OnyxException(message) {

    companion object {
        const val ATTRIBUTE_TYPE_IS_NOT_SUPPORTED = "Attribute type is not supported"
    }
}
